Software/Autonomous/Autonomous.ino

```diff
@@ -0,0 +1,88 @@
+//Bibliotecas internas
+#include "_config.h"
+#include "constants_declaration.h"
+#include "controller_IR.h"
+#include "motors.h"
+#include "distance_sensors.h"
+#include "utils.h"
+
+int flag = 0; //Direita --> 1, Esquerda --> -1, Valor inicial --> 0
+unsigned long contador = millis();
+
+void setup(){
+  Serial.begin(115200);
+
+  controllerInit();
+  sensorsInit();
+  motorsInit();
+}
+
+bool needsToStop(bool velEsqPositivaNova, bool velDirPositivaNova ) {
+  if (speedL == 0 and speedR == 0) return false; //Se a vel é 0, pode ir pra frente ou trás sem parar
+
+  speedLPositiva = speedL > 5 ? true : false; //Adicionei uma margem para ficar de acordo com a de 84 ate 94 que vai pra esc
+  speedRPositiva = speedR > 5 ? true : false;
+  if ((speedLPositiva == velEsqPositivaNova) and (speedRPositiva == velDirPositivaNova)) {
+    if (contadorFlutuacoes > 0) contadorFlutuacoes--; //Pode ser que a gnt soh teve variacoes temporarias, entao a gnt vai tirando elas
+    return false; //Se o sentido q vai rodar é igual ao anterior, nao para!
+  }
+
+  contadorFlutuacoes++;
+  if (contadorFlutuacoes >= 10) {
+    speedL = speedR = 0; // APenas para caso a gnt tenha tido flutuacoes seguidas o suficiente. 10 eh o bastante?
+    motorsOutput();
+    delay(100); //TODO: Podemos aumentar esse valor p/ ver no codigo mais claramente se ele para antes de mudar a direcao
+    contadorFlutuacoes = 0;
+    return false; //Retorna falso para o codigo nao fazer um return dentro do loop!
```

Comment:
Não entendi esse comentário `Retorna falso para o codigo nao fazer um return dentro do loop!`. Retornar falso é um `return`de qualquer forma. Clarifiquem essa parte por favor.

Reply:
Perceba que o needsToStop n está só tacado no código em uma linha sozinho, mas ele está dentro de um if! Se a gente retorna true, nós temos um "return dentro da função loop() do arduíno". E se ele retorna false, o código segue rodando como se nada tivesse acontecido. Por que fazer isso? Para eu não mudar a velocidade se não for necessário!

![image](https://user-images.githubusercontent.com/79096190/227741557-cffda60f-d41c-407c-88a0-ffa7ab25f11f.png)

Aqui por exemplo:

![image](https://user-images.githubusercontent.com/79096190/227741639-eefc8929-40c1-4126-8e31-33e9341cccef.png)

Se eu não fizesse um return logo após o if, o loop ia seguir e eu teria a minha velocidade sendo mudada mesmo não sendo isso o que eu desejava! Então o true ou false que a função retorna é justamente para impedir que uma velocidade nova indesejada seja adicionada sem que eu antes pare o motor

Reply:
Essa observação foi em relação ao comentário, que não está claro o suficiente na minha opinião.